### PR TITLE
[Added] Coin Change problem with solution Greedy Algorithm

### PR DESCRIPTION
**Pull request by FrejBjornsson**
Please include a summary of the change. Please also include relevant motivation and context (if applicable).

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the comments
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Security.Cryptography;
+using System.Text;
+
+public class GreedyAlgorithm
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:6]: Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- Algorithms/Problems/CoinChange/GreedyAlgorithm.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770&bid=20623044)

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    public static void Main()
+    {
+            Console.WriteLine("Enter the amount of cents you want to change (e.g. 0.64):");
+            double origAmount = Convert.ToDouble(Console.ReadLine());
+            double toChange = origAmount;
+            double remainAmount = 0.0;
+            int[] coins = new int[4];
+            MakeChange(origAmount, remainAmount, coins);
+            Console.WriteLine("The best way to change " +
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:77]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        if (arr[1] > 0)
+        {
+            Console.WriteLine("Number of nickels: " + arr[1]);
+        }
+
+        if (arr[0] > 0)
+        {
+            Console.WriteLine("Number of pennies: " + arr[0]);
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:65]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            coins[0] = (int)(origAmount / 0.01);
+            remainAmount = origAmount % 0.01;
+        }
+    }
+
+    public static void ShowChange(int[] arr)
+    {
+        if (arr[3] > 0)
+        {
+            Console.WriteLine("Number of quarters: " + arr[3]);
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:50]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        if (arr[2] > 0)
+        {
+            Console.WriteLine("Number of dimes: " + arr[2]);
+        }
+
+        if (arr[1] > 0)
+        {
+            Console.WriteLine("Number of nickels: " + arr[1]);
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:60]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+        if (arr[0] > 0)
+        {
+            Console.WriteLine("Number of pennies: " + arr[0]);
+        }
+    }
+
+    public static void Main()
+    {
+            Console.WriteLine("Enter the amount of cents you want to change (e.g. 0.64):");
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:71]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+using System.Linq.Expressions;
+using System.Security.Cryptography;
+using System.Text;
+
+public class GreedyAlgorithm
+
+    /*
+    // Info
+    //
+A greedy algorithm is the one that always chooses the best solution at the time, with no regard for how that choice will affect future choices. Here, we will discuss how to use Greedy algorithm to making coin changes.
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:11]: Codacy found an issue: [Split this 217 characters long line (which is greater than 200 authorized).](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    public static void ShowChange(int[] arr)
+    {
+        if (arr[3] > 0)
+        {
+            Console.WriteLine("Number of quarters: " + arr[3]);
+        }
+
+        if (arr[2] > 0)
+        {
+            Console.WriteLine("Number of dimes: " + arr[2]);
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:55]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

File: Algorithms/Problems/CoinChange/GreedyAlgorithm.cs [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+using System.Text;
+
+public class GreedyAlgorithm
+
+    /*
+    // Info
+    //
+A greedy algorithm is the one that always chooses the best solution at the time, with no regard for how that choice will affect future choices. Here, we will discuss how to use Greedy algorithm to making coin changes.
+It has been proven that an optimal solution for coin changing can always be found using the current American denominations of coins
+For an example, Let’s say you buy some items at the store and the change from your purchase is 63 cents. How does the clerk determine the change to give you? If the clerk follows a greedy algorithm, he or she gives you two quarters, a dime, and three pennies. That is the smallest number of coins that will equal 63 cents.
```
Comment (siriak) [inline Algorithms/Problems/CoinChange/GreedyAlgorithm.cs:13]: Codacy found an issue: [Split this 322 characters long line (which is greater than 200 authorized).](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6350770)

**Comment by FrejBjornsson** [comment]
Lol what

**Review by siriak — COMMENTED** [review]
Hi! Take a look at how algorithms are organized in this repo, we don't write output to console and don't include the Main() method. Unit tests are required too

**Comment by FrejBjornsson** [comment]
Ok thanks

**Pull request closed without merging**